### PR TITLE
Creating CRUD methods and adding tests for them

### PR DESCRIPTION
**Pull request by OlegDjur**

### CODE REVIEW

File: postgres/user.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+	"github.com/jmoiron/sqlx"
+)
+
+type UserStore struct {
+	*sqlx.DB
+}
+
+func NewUserStore(db *sqlx.DB) *UserStore {
+	return &UserStore{
```
[the remark this answers is not in the document]
Comment (OlegDjur) [reply postgres/user.go:15]: why don't I need a constructor? Do the methods need to be done specifically for our users structure?
in this case, I need to pass an object db  inside the method. if I pass db object as an argument, is it ok?

[the remark this answers is not in the document]
Comment (KlaraBauyrzhanova) [reply postgres/user.go:15]: Try to use:
```go
func init()
```
It will help:)

File: db/db.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	}
+	DB = db
+	m, err := migrate.New("file://migrate/", connStr)
+	if err != nil {
+		log.Fatalf("error migration: %v", err)
+	}
+
+	if err := m.Up(); err != nil && err != migrate.ErrNoChange {
+		log.Fatalf("error migration up: %v", err)
+	}
```
Comment (KlaraBauyrzhanova) [inline db/db.go:38]: I suggest the following:
```go
func init() {
    DB = newDB()
}

func newDB() *sqlxDB {
	connStr := "postgres://" + userName + ":" + password + "@" + host + ":" + port + "/" + dbname + "?" + "sslmode=disable"
	db, err := sqlx.Connect("postgres", connStr)
	if err != nil {
		log.Fatalf("error to connect database: %v", err)
	}
	m, err := migrate.New("file://migrate/", connStr)
	if err != nil {
		log.Fatalf("error migration: %v", err)
	}

	if err := m.Up(); err != nil && err != migrate.ErrNoChange {
		log.Fatalf("error migration up: %v", err)
	}
    return db
}

File: pkg/users/users.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+func NewUser(data string) *User {
+	return &User{
+		Data: data,
+	}
+}
+
+func (s *User) Create() error {
+	_, err := db.DB.NamedExec(`INSERT INTO users (data) VALUES (:data)`,
+		map[string]interface{}{
```
Comment (KlaraBauyrzhanova) [inline pkg/users/users.go:22]: Where is `id`?

File: pkg/users/users.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+			"data": s.Data,
+		})
+	return err
+}
+
+func (s *User) Read() {
+	rows, err := db.DB.NamedQuery(`SELECT * FROM users WHERE data=:fn`, map[string]interface{}{"fn": s.Data})
```
Comment (KlaraBauyrzhanova) [inline pkg/users/users.go:29]: Are you gonna look for by `data`?
I guess it isn't correct.

File: pkg/users/users.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+		err := rows.StructScan(&s)
+		if err != nil {
+			log.Fatalln(err)
+		}
+		fmt.Println(s)
+	}
+}
+
+func (s *User) Update(data string) error {
+	_, err := db.DB.NamedExec(`UPDATE users SET data=:data WHERE data=:userdata`,
```
Comment (KlaraBauyrzhanova) [inline pkg/users/users.go:43]: I guess it doesn't work.

File: pkg/users/users.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+		map[string]interface{}{
+			"data":     data,
+			"userdata": s.Data,
+		})
+	return err
+}
+
+func (s *User) Delete() error {
+	if _, err := db.DB.NamedExec(`DELETE FROM users WHERE data =:data`,
```
Comment (KlaraBauyrzhanova) [inline pkg/users/users.go:52]: Are you sure that is it correct? I am not sure deleting value by `data` is right thing.

**Pull request merged**